lib/cext/preprocess.rb

```diff
@@ -191,6 +434,7 @@ def preprocess(line)
 end
 
 def patch(file, contents, directory)
+  file = File.basename(file)
```

Comment:
This is to handle filenames like: `../../../../ext/mysql2/client.c`.

Reply:
Seems OK, although I would move it to the line below so it's clear it's only used for looking up in PATCHED_FILES:
`PATCHED_FILES[File.basename(file)]`

Reply:
Fixed.